### PR TITLE
Fix additional mobile signon bugs

### PR DESCRIPTION
**Pull request by sddioulde**
### Description

Fix additional mobile signon bugs

### Dragons

*Is there anything the reviewer should be on the lookout for? Are there any dangerous changes?*

### How Has This Been Tested?

Locally spin up ios and android emulators

### How will this change be monitored?

*For features that are critical or could fail silently please describe the monitoring/alerting being added.*

### CODE REVIEW

File: src/components/signon/SignOn.tsx [hunk 1]
```diff
@@ -321,6 +324,7 @@ const SignOn = ({ navigation }: SignOnProps) => {
   const emailIsAvailable = useSelector(getEmailIsAvailable)
   const emailIsValid = useSelector(getEmailIsValid)
   const emailStatus = useSelector(getEmailStatus)
+  const isKeyboardVisible = useSelector(getIsKeyboardVisible)
```
Comment (sliptype) [inline src/components/signon/SignOn.tsx:327]: I think you can use the existing selector for this:
https://github.com/AudiusProject/audius-mobile-client/blob/master/src/store/keyboard/selectors.ts#L4

And then you won't need to make any changes to `useKeyboardListeners`

Comment (sddioulde) [reply src/components/signon/SignOn.tsx:327]: Oh nice! Thanks

File: src/components/notifications/Notifications.tsx [hunk 2]
```diff
@@ -291,7 +291,7 @@ const Notifications = ({
 
   const panProps = canShowNotifications ? panResponder.panHandlers : {}
 
-  return (
+  return isSignedIn ? (
```
Comment (raymondjacobson) [inline src/components/notifications/Notifications.tsx:294]: is this true when the session first loads? (i think probably, but just want to make sure notifs are accessible right away still)

File: src/store/keyboard/sagas.ts [hunk 3]
```diff
@@ -1,11 +1,30 @@
...
+    Keyboard.addListener('keyboardDidShow', () => {
+      emitter('show')
+    })
+    Keyboard.addListener('keyboardDidHide', () => {
+      emitter('hide')
+    })
+    return () => {}
+  })
+
+  yield fork(function* () {
```
Comment (raymondjacobson) [inline src/store/keyboard/sagas.ts:18]: what does the fork gain here? probably can just put the while (true) { } in line right?

File: src/store/sagas.ts [hunk 4]
```diff
@@ -1,5 +1,10 @@
...
+import { all, call } from 'redux-saga/effects'
+import initKeyboardEvents from './keyboard/sagas'
 
 export default function* rootSaga() {
-  return [initKeyboardEvents]
+  // executing array of effects was deprecated in v1 of redux-saga
+  // must call them in the all effect now
+  // https://stackoverflow.com/questions/55811145/redux-saga-does-not-run/55811902#55811902
+  const sagas = [initKeyboardEvents]
+  yield all(sagas.map(call))
```
Comment (raymondjacobson) [inline src/store/sagas.ts:9]: https://github.com/AudiusProject/audius-client/blob/master/src/store/sagas.js#L149

**Review by raymondjacobson — APPROVED** [review]

File: src/store/signon/reducer.ts [hunk 5]
```diff
@@ -39,6 +39,7 @@ export type SignonState = {
   }
   finalEmail: string
   finalHandle: string
+  isKeyboardVisible: boolean
```
Comment (sliptype) [inline src/store/signon/reducer.ts:42]: This can be removed now

File: src/components/notifications/Notifications.tsx (hunk 2 above)
Comment (sddioulde) [reply src/components/notifications/Notifications.tsx:294]: yes it is

**Pull request merged**